meson.build

```diff
@@ -51,7 +51,7 @@ git = find_program('git', required: true)
 init = custom_target(
   'fake_init',
   output: ['init.c'],
-  command: [python3, meson.source_root() + '/tools/generate_init', '-o', '@OUTPUT0@'],
+  command: [python3, meson.source_root() + '/tools/generate_init', '-o', '@OUTPUT0@', '-d', meson.current_build_dir()],
   build_always_stale: true,
```

Comment:
Why do we need to explicitly set this? The arguments are a bit counterintuitive as there are now two switches to control the output path.

Reply:
Because without it, the 'init.c' file is generated, but in the wrong directory (somewhere in the source directory).
The output path is defined by the meson build directory option at setup phase.

I got several times this error :
> [8/201] x86_64-pc-linux-gnu-gcc -ITR1X.p -I. -I../TR1X-9999 -I../TR1X-9999/src -I/usr/include/SDL2 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -Wno-unused -DMESON_BUILD -march=x86-64 -mtune=generic -O2 -pipe -mmmx -msse -msse2 -msse3 -mssse3 -msse4.1 -msse4.2 -mcx16 -mfxsr -mpclmul -mpopcnt -msahf -mavx -mxsave -mxsaveopt -fstack-protector-strong -mno-avx2 -mno-sse4a -mno-fma4 -mno-xop -mno-fma -mno-avx512f -mno-bmi -mno-bmi2 -mno-aes -mno-avx512vl -mno-avx512bw -mno-avx512dq -mno-avx512cd -mno-avx512er -mno-avx512pf -mno-avx512vbmi -mno-avx512ifma -mno-avx5124vnniw -mno-avx5124fmaps -mno-avx512vpopcntdq -mno-avx512vbmi2 -mno-gfni -mno-vpclmulqdq -mno-avx512vnni -mno-avx512bitalg -mno-avx512bf16 -mno-avx512vp2intersect -mno-3dnow -mno-adx -mno-abm -mno-cldemote -mno-clflushopt -mno-clwb -mno-clzero -mno-enqcmd -mno-f16c -mno-fsgsbase -mno-hle -mno-lwp -mno-lzcnt -mno-movbe -mno-movdir64b -mno-movdiri -mno-mwaitx -mno-pconfig -mno-pku -mno-prefetchwt1 -mno-prfchw -mno-ptwrite -mno-rdpid -mno-rdrnd -mno-rdseed -mno-rtm -mno-serialize -mno-sgx -mno-sha -mno-shstk -mno-tbm -mno-tsxldtrk -mno-vaes -mno-waitpkg -mno-wbnoinvd -mno-xsavec -mno-xsaves -mno-amx-tile -mno-amx-int8 -mno-amx-bf16 -mno-uintr -mno-hreset -mno-kl -mno-widekl -mno-avxvnni -mno-avx512fp16 -mno-avxifma -mno-avxvnniint8 -mno-avxneconvert -mno-cmpccxadd -mno-amx-fp16 -mno-prefetchi -mno-raoint -mno-amx-complex -D_REENTRANT -MD -MQ TR1X.p/meson-generated_.._init.c.o -MF TR1X.p/meson-generated_.._init.c.o.d -o TR1X.p/meson-generated_.._init.c.o -c init.c
FAILED: TR1X.p/meson-generated_.._init.c.o 
x86_64-pc-linux-gnu-gcc -ITR1X.p -I. -I../TR1X-9999 -I../TR1X-9999/src -I/usr/include/SDL2 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -Wno-unused -DMESON_BUILD -march=x86-64 -mtune=generic -O2 -pipe -mmmx -msse -msse2 -msse3 -mssse3 -msse4.1 -msse4.2 -mcx16 -mfxsr -mpclmul -mpopcnt -msahf -mavx -mxsave -mxsaveopt -fstack-protector-strong -mno-avx2 -mno-sse4a -mno-fma4 -mno-xop -mno-fma -mno-avx512f -mno-bmi -mno-bmi2 -mno-aes -mno-avx512vl -mno-avx512bw -mno-avx512dq -mno-avx512cd -mno-avx512er -mno-avx512pf -mno-avx512vbmi -mno-avx512ifma -mno-avx5124vnniw -mno-avx5124fmaps -mno-avx512vpopcntdq -mno-avx512vbmi2 -mno-gfni -mno-vpclmulqdq -mno-avx512vnni -mno-avx512bitalg -mno-avx512bf16 -mno-avx512vp2intersect -mno-3dnow -mno-adx -mno-abm -mno-cldemote -mno-clflushopt -mno-clwb -mno-clzero -mno-enqcmd -mno-f16c -mno-fsgsbase -mno-hle -mno-lwp -mno-lzcnt -mno-movbe -mno-movdir64b -mno-movdiri -mno-mwaitx -mno-pconfig -mno-pku -mno-prefetchwt1 -mno-prfchw -mno-ptwrite -mno-rdpid -mno-rdrnd -mno-rdseed -mno-rtm -mno-serialize -mno-sgx -mno-sha -mno-shstk -mno-tbm -mno-tsxldtrk -mno-vaes -mno-waitpkg -mno-wbnoinvd -mno-xsavec -mno-xsaves -mno-amx-tile -mno-amx-int8 -mno-amx-bf16 -mno-uintr -mno-hreset -mno-kl -mno-widekl -mno-avxvnni -mno-avx512fp16 -mno-avxifma -mno-avxvnniint8 -mno-avxneconvert -mno-cmpccxadd -mno-amx-fp16 -mno-prefetchi -mno-raoint -mno-amx-complex -D_REENTRANT -MD -MQ TR1X.p/meson-generated_.._init.c.o -MF TR1X.p/meson-generated_.._init.c.o.d -o TR1X.p/meson-generated_.._init.c.o -c init.c
cc1: fatal error: init.c: No such file or directory
compilation terminated.

Reply:
Thanks, this makes sense. Any way to use a single argument for the output path? Does meson allow concatenating special variables such as `@OUTPUT0@`?

Reply:
> The arguments are a bit counterintuitive as there are now two switches to control the output path.

Ah you're right, I see your point. I will simplify.

Reply:
> > The arguments are a bit counterintuitive as there are now two switches to control the output path.
> 
> Ah you're right, I see your point. I will simplify.

I'm afraid I have no choice here.
The meson custom_target 'output' keyword can not have a path segment.
If I try to do this :
`
init = custom_target(
  'fake_init',
   output: [meson.current_build_dir() / 'init.c'], [...]
`

meson fails with :
> meson.build:51:7: ERROR:  custom_target keyword argument "output" Output '/var/tmp/portage/games-engines/TR1X-9999/work/TR1X-9999-build/init.c' must not contain a path segment.

Reply:
> Does meson allow concatenating special variables such as `@OUTPUT0@`?

Yes, you're right. Simply do it on the good place. That's done.